src/app/upgrade.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2015  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#ifndef UPGRADE_H
+#define UPGRADE_H
+
+#include <libtorrent/lazy_entry.hpp>
+#include <libtorrent/entry.hpp>
+#include <libtorrent/bencode.hpp>
+
+#include <QString>
+#include <QDir>
+#include <QFile>
+#ifndef DISABLE_GUI
+#include <QMessageBox>
+#endif
+
+#include "core/utils/fs.h"
+#include "core/utils/misc.h"
+
+bool userAcceptsUpgrade()
+{
+#ifdef DISABLE_GUI
+    std::cout << std::endl << "*** " << qPrintable(QObject::tr("Upgrade")) << " ***" << std::endl;
+    std::cout << qPrintable(QObject::tr("You updated from an older version that saved things differently. You must migrate to the new saving system. If you answer Yes, you will not be able to go back and use an older version than v3.3.0.")) << std::endl << std::endl;
+    std::cout << qPrintable(QObject::tr("Press %1 key to accept and continue...").arg("'y'")) << std::endl;
+    char ret = getchar(); // Read pressed key
+    if (ret == 'y' || ret == 'Y')
+        return true;
+#else
+    QMessageBox msgBox;
+    msgBox.setText(QObject::tr("You updated from an older version that saved things differently. You must migrate to the new saving system. If you click [Continue], you will not be able to go back and use an older version than v3.3.0."));
```

Comment:
`You updated from an older version that saved things differently. You must migrate to the new saving system. If you continue, you will not be able to use an older version than v3.3.0 again.`